README.md

```diff
@@ -276,81 +278,383 @@ schedule is listed as N/A.
+in-memory size of the table.
+
+Tables maintain a [Bloom
+filter](https://en.wikipedia.org/wiki/Bloom_filter "https://en.wikipedia.org/wiki/Bloom_filter")
+in memory for each run on disk. These Bloom filter are are
+probablilistic datastructure that is used to track which keys are
+present in their corresponding run. Querying a Bloom filter returns
+either "maybe" meaning the key is possibly in the run or "no" meaning
+the key is definitely not in the run. When a query returns "maybe" while
+the key is *not* in the run, this is referred to as a *false positive*.
+While the database executes a lookup operation, any Bloom filter query
+that returns a false positive causes the database to unnecessarily read
+a run from disk. The probabliliy of these spurious reads follow a
+[binomial
+distribution](https://en.wikipedia.org/wiki/Binomial_distribution "https://en.wikipedia.org/wiki/Binomial_distribution")
+$`\operatorname{Binomial}(r,\operatorname{FPR})`$ where $`r`$ refers to
+the number of runs and $`\operatorname{FPR}`$ refers to the
+false-positive rate of the Bloom filters. Hence, the expected number of
+spurious reads for each lookup operation is
+$`r\cdot\operatorname{FPR}`$. The number of runs $`r`$ is proportional
+to the number of physical entries in the table. Its exact value depends
+on the merge policy of the table:
+
+`LazyLevelling`  
+$`r = T (\log_T\frac{n}{B} - 1) + 1`$.
+
+The false-positive rate scales exponentially with size of the Bloom
+filters in bits per entry.
+
+| False-positive rate (FPR) | Bits per entry (BPE) |
+|---------------------------|----------------------|
+| $`1\text{ in }10`$        | $`\approx  4.77 `$   |
+| $`1\text{ in }100`$       | $`\approx  9.85 `$   |
+| $`1\text{ in }1{,}000`$   | $`\approx 15.78 `$   |
+| $`1\text{ in }10{,}000`$  | $`\approx 22.57 `$   |
+| $`1\text{ in }100{,}000`$ | $`\approx 30.22 `$   |
+
+The configuration parameter `confBloomFilterAlloc` can be specified in
+two ways:
+
+`AllocFixed bitsPerEntry`  
+Allocate the requested number of bits per entry in the table.
+
+The value must strictly positive, but fractional values are permitted.
+The recommended range is $`[2, 24]`$.
+
+`AllocRequestFPR falsePositiveRate`  
+Allocate the required number of bits per entry to get the requested
+false-positive rate.
+
+The value must be in the range $`(0, 1)`$. The recommended range is
+$`[1\mathrm{e}{ -5 },1\mathrm{e}{ -2 }]`$.
+
+The total in-memory size of all Bloom filters scales *linearly* with the
+number of physical entries in the table and is determined by the number
+of physical entries multiplied by the number of bits per physical entry,
+i.e, $`n\cdot\operatorname{BPE}`$. For example, a table with 100 million
+physical entries and an FPR of 1 in 1,000 will allocate 188MiB of memory
+for Bloom filters. If the table uses lazy levelling, a size ratio of 4,
+and a write buffer capacity of 20,000 entries, then a spurious read can
+be expected once every 46 lookup operations.
```

Comment:
Did we cover how this overall FPR is calculated? I missed it.

Reply:
It's slightly further up, where I state that it's a binomial distribution. The expected value of `B(n,p)` is simply `np` so there's not much to explain.